### ISSUE TITLE
Target .Net Core or .Net Standard

### ISSUE THREAD

**Issue by CodeMyst** [marshallward/TiledSharp#54]
From what I've seen while using this project, it works perfectly fine with .Net Core (2.1). I think it would be great to make the project target .Net Standard (2.0) so that the package can be used in both .Net Framework and .Net Core projects.

I can try and give this a shot, shouldn't be too difficult to do.

**Comment by marshallward** [comment]
Sure, a few people have also considered looking into this, it might also be helpful to look over some of the older issues.

**Comment by CodeMyst** [comment]
So far I managed to convert the project to use .Net Standard 2.0 which can be used in .Net Core and .Net Framework projects. You can see the full platform support here: [.Net Standard](https://docs.microsoft.com/en-us/dotnet/standard/net-standard).

The project is on here: https://github.com/CodeMyst/TiledSharp/tree/net-standard

I was wondering why in the README.md it says it uses the library DotNetZip but I couldn't find it being used in the actual project.

The tests can be run from VS or if using the command line (and .Net Core) with
```
dotnet test --framework netcoreapp2.1
```
or
```
dotnet test --framework net40
```
and so on.

**Comment by marshallward** [comment]
DotNetZip was removed some time ago, since I was able to get zlib support by stripping the header and using .NET intrinsics.  So the README is just out of date :).  I've just pushed a change removing these references.

The changes seem very good, thanks for adding the tests and the DiagonalTest flag.  It seems like you've checked that .NET core still works.  I don't really follow this myself since I'm not very involved in this project anymore, but I seem to get pull requests from others to support it.

One thing it would be nice to confirm is that things run under net35 since this seems to be what Unity still uses (although I continuously hear about "updates" to their .NET support, which is confusing).  This was, by the way, the biggest reason for removing DotNetZip.

I'll try to get familiar with the tests you provided but in general this seems good!

**Comment by CodeMyst** [comment]
Thanks!

I added support for .Net Framework 3.5 right now. Although that is only necessary for Unity 2018.1 and older, since then Unity has .Net Framework 4.x support. And I tested that both the 3.5 and 4.5 versions work in Unity and it seems pretty good.

Next on the list is getting the nuget package building to work. I have 0 knowledge for how nuget packages are created so gonna have to research that a bit.

**Comment by CodeMyst** [comment]
NuGet now also works. I tested it on a .Net Core 2.1 and .Net Framework 4.5 project and the NuGet package uses the appropriate dll file according to the project target framework.

I'll create a pull request for now.

**Comment by marshallward** [comment]
Great to hear, I am away this weekend but will give it a try next week.

**Comment by CodeMyst** [comment]
Closed by #55
